### PR TITLE
Feature/remove spock

### PR DESCRIPTION
**Pull request by mikeSmith08**
As well as replacing the Spock tests, there was duplication with the controller integration tests, so that has been removed. Also missed an integration test where the Eligibility Service call fails, so have added that. This PR replaces the Spock smoke test too.

### CODE REVIEW

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    private void assertClaimantPersistedSuccessfully(ClaimantDTO claimantDTO, EligibilityStatus eligibilityStatus) {
+        Iterable<Claimant> claims = claimantRepository.findAll();
+        assertThat(claims).hasSize(1);
+        Claimant persistedClaim = claims.iterator().next();
+        assertClaimantMatchesClaimantDTO(claimantDTO, persistedClaim);
+        assertThat(persistedClaim.getEligibilityStatus()).isEqualTo(eligibilityStatus);
+    }
+
+    @Test
+    void shouldFailWhenEligibilityServiceCallThrowsException() {
```
Comment (dwybourn) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java]: This and the test below should probably be in ClaimantServiceIntegrationErrorTests

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        ClaimDTO claim = aValidClaimDTO();
+        modifyFieldOnObject(claim.getClaimant().getCardDeliveryAddress(), fieldName, value);
+        //When
+        ResponseEntity<ErrorResponse> response = restTemplate.exchange(buildRequestEntity(claim), ErrorResponse.class);
+        //Then
+        assertValidationResponse(response, expectedField, expectedErrorMessage);
+    }
+
+    @ParameterizedTest(name = "Field {0} with invalid value {1} on an error response")
+    @MethodSource("shouldFailWithInvalidDateFormatErrorArguments")
```
Comment (dwybourn) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java]: CsvSource might be better here as the test data will be above the test class. Uses less code as well.
E.g.
```
@CsvSource({
            "1999/12/31, dateOfBirth, '1999/12/31' could not be parsed as a LocalDate, claimant.dateOfBirth",
            "Foo, dateOfBirth, 'Foo' could not be parsed as a LocalDate, claimant.dateOfBirth",
            "29-11-1909, expectedDeliveryDate, '29-11-1909' could not be parsed as a LocalDate, claimant.expectedDeliveryDate",
            "1999/12/31, expectedDeliveryDate, '1999/12/31' could not be parsed as a LocalDate, claimant.expectedDeliveryDate",
            "Foo, expectedDeliveryDate, 'Foo' could not be parsed as a LocalDate, claimant.expectedDeliveryDate"
    })
```

**Comment by dwybourn** [comment]
Build currently fails as the ErrorResponse imports need importing.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java (hunk 2 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java]: I think I'd go for the csv option too.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/AbstractIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import uk.gov.dhsc.htbhf.claimant.model.AddressDTO;
+import uk.gov.dhsc.htbhf.claimant.model.ClaimantDTO;
+
+import java.net.URI;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+public abstract class AbstractIntegrationTest {
```
Comment (YetAnotherMatt) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/AbstractIntegrationTest.java]: Something to consider: How about making this class static and using static imports on the test classes? Then you just need to move the restTemplate back into the test classes (where it is probably better situated anyway)

Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/AbstractIntegrationTest.java]: Yes, true, this was a better fit when I originally did this, but I've moved more and more out so a static utility class is now a better fit.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java (hunk 1 above)
Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java]: As discussed, we'll keep it in here as that class mocks out the `ClaimService`.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java]: Ok, I'm happy with either but I'll change it as you've both commented on it.

**Pull request merged**